torch/csrc/Generator.cpp

```diff
@@ -47,15 +52,22 @@ static PyObject * THPGenerator_pynew(PyTypeObject *type, PyObject *args, PyObjec
   torch::ParsedArgs<1> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
   auto device = r.deviceWithDefault(0, at::Device(at::kCPU));
+  TORCH_CHECK(device.type() == at::kCPU || device.type() == at::kCUDA,
```

Comment:
Checking here is insufficient: you will accept CUDA even on a non-CUDA build, and then you'll return a CPU generator due to how the ifdef is setup.